src/EditorFeatures/CSharpTest/CompleteStatement/CSharpCompleteStatementCommandHandlerTests.cs

```diff
@@ -0,0 +1,2068 @@
+using VSCommanding = Microsoft.VisualStudio.Commanding;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CompleteStatement
+{
+    public class CSharpCompleteStatementCommandHandlerTests : AbstractCompleteStatementTests
+    {
+        private string CreateTestWithMethodCall(string code)
+        {
+            return
+@"class C
+    {
+        static void Main(string[] args)
+        {
+            int x = 1;
+            int y = 2;
+            int[] a = { 1,2 }
+            " + code + @"
+
+            int z = 4;
+        }
+    }
+
+    static class ClassC
+    {
+        internal static int MethodM(int a, int b)
+            => a * b;
+    }
+}";
+        }
+
+        #region ArgumentListOfMethodInvocation
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfMethodInvocation1()
+        {
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM($$x, y)");
+
+            var expected = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y);$$");
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfMethodInvocation2()
+        {
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x$$, y)");
+
+            var expected = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y);$$");
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfMethodInvocation3()
+        {
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x,$$ y)");
+
+            var expected = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y);$$");
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfMethodInvocation4()
+        {
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, $$y)");
+
+            var expected = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y);$$");
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfMethodInvocation5()
+        {
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y$$)");
+
+            var expected = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y);$$");
+
+            VerifyTypingSemicolon(code, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.CompleteStatement)]
+        public void ArgumentListOfMethodInvocation6()
+        {
+            var code = CreateTestWithMethodCall(@"var test = ClassC.MethodM(x, y)$$");
+
+            var expected = code.Replace("$$", ";$$");
```

Comment:
from our offline convo, this is being used to assert "the semicolon is inserted as it would have been before."  But i'd *rather* have a specific test method that validates that.  i.e. `VerifyNoSpecialSemicolonHandling(".$$..");`.  This could then run the scenario both with and without htat command handler and it can verify the results are the same.

Reply:
> `VerifyNoSpecialSemicolonHandling`

👍

Reply:
Note: as per previous comment, this would also go through the whole stack of formatting/brace-completion/completion/complete-statement just to really verify nothing was getting screwy in that.

Reply:
I am not aware of a way to test the full stack accurately. I'll check with the editor on this but leaving it as a follow-up issue.